abel/hansenlaw.py

```diff
@@ -103,33 +101,28 @@ class :class:``abel.Transform``.
     """
 
     IM = np.atleast_2d(IM)
-    N = np.shape(IM)         # shape of input quadrant (half)
-    AIM = np.zeros(N)        # forward/inverse Abel transform image
-
-    rows, cols = N
+    rows, cols = np.shape(IM)      # shape of input quadrant (half)
+    AIM = np.zeros_like(IM)        # forward/inverse Abel transform image
 
     # Two alternative Gamma functions for forward/inverse transform
     # Eq. (16c) used for the forward transform
-    def fgamma(Nm, lam, n):
-        return 2*n*(1-np.power(Nm, lam+1))/(lam+1)
+    def fgamma(N, lam, n):
+        lam += 1
+        return 2*n[::-1]*(1 - N**lam)/lam
 
     # Eq. (18) used for the inverse transform
-    def igammalt(Nm, lam, n):
-        return (1-np.power(Nm, lam))/(pi*lam)
+    def igammalt(N, lam, n):
+        return (1 - N**lam)/(np.pi*lam)
```

Comment:
Woah, it turns out that `numpy.power` can be significantly slower than `**`. Good to know!

https://stackoverflow.com/questions/25870923/how-to-square-or-raise-to-a-power-elementwise-a-2d-numpy-array

Reply:
What about np.square(x) as one of the comments in the SO link states?